tests/unit/blocking_connection_tests.py

```diff
@@ -5,6 +5,10 @@
 """
 import socket
 
+import errno
+
+from pika.exceptions import ConnectionClosed
```

Comment:
A commonly-used practice is to group imports as follows for readability:

1. Imports of builtins
2. Imports of 3rd party modules
3. Imports of our own modules

So, this new pika.exceptions import belongs with the other pika imports below

Also, all other code in this test suite accesses pika exceptions relative to their namespace, so to follow that convention, your code would access it as `pika.exceptions.ConnectionClosed`. However, we still need an explicit import `import pika.exceptions` after `from pika.adapters import ...`. The original implementation forgot to add that import and got by implicitly.